### PR TITLE
Add `chains` to all `Input` classes

### PR DESCRIPTION
**Pull request by amorehead**
* Adds `chains` to all `Input` classes
* Next steps include wiring the `chains` into how we select which val/test metrics to report for which (sampled) chains. In principle, we could just subsample the chain IDs for which we report train/val/test metrics uniformly, since cropping chains then subsampling metrics by the same (cropped) chain IDs should behave the same as not cropping chains then subsampling the metrics.
* In other words, I think we should just tag a small bit of logic in our metric reporting code that filters down the final reported metric values to only those chains that are denoted by `chains`. Also, if `chains` is `None`, I think we should just (like normal) report metric values for all chains in the input structure(s).

### CODE REVIEW

File: alphafold3_pytorch/inputs.py [hunk 1]
```diff
@@ -168,6 +168,7 @@ class AtomInput:
     pde_labels:                 Int['n n'] | None = None
     plddt_labels:               Int[' n'] | None = None
     resolved_labels:            Int[' n'] | None = None
+    chains:                     Tuple[int | None, int | None] = (None, None)
```
Comment (lucidrains) [inline alphafold3_pytorch/inputs.py:171]: a type of `Int[' 2']` tensor at the `AtomInput` would lead to less issues, and `Int['b 2']` on `BatchAtomInput`

Comment (lucidrains) [reply alphafold3_pytorch/inputs.py:171]: for `None`, can just use something out of range, like say `-1`

Comment (amorehead) [reply alphafold3_pytorch/inputs.py:171]: Sg

Comment (amorehead) [reply alphafold3_pytorch/inputs.py]: Ah, also got to update the forward()

Comment (amorehead) [reply alphafold3_pytorch/inputs.py]: And I also need to listen to myself and make `chains` optional :)

**Comment by lucidrains** [comment]
@amorehead 💯 🚀
